modules_core/com.etendoerp.legacy.advpaymentmngt/src/com/etendoerp/legacy/utilitySequence/UtilitySequenceLegacy.java

```diff
@@ -16,167 +16,204 @@
-                                                 String AD_Client_ID, String TableName, boolean updateNext) {
-        if (TableName == null || TableName.length() == 0) {
-            throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
-        }
-
-        CSResponse cs = null;
-        try {
-            cs = DocumentNoData.nextDocConnection(conn, con, "DocumentNo_" + TableName, AD_Client_ID,
-                    (updateNext ? "Y" : "N"));
-        } catch (final ServletException e) {
-        }
-
-        if (cs == null || cs.razon == null) {
-            return "";
-        } else {
-            return cs.razon;
-        }
+  /**
+   * Gets the document number from the database.
+   *
+   * @param conn
+   *     Handler for the database connection.
+   * @param vars
+   *     Handler for the session info.
+   * @param WindowNo
+   *     Window id.
+   * @param TableName
+   *     Table name.
+   * @param C_DocTypeTarget_ID
+   *     Id of the doctype target.
+   * @param C_DocType_ID
+   *     id of the doctype.
+   * @param onlyDocType
+   *     Search only for doctype.
+   * @param updateNext
+   *     Save the new sequence in database.
+   * @return String with the new document number.
+   */
+  @Override
+  public String getDocumentNo(ConnectionProvider conn, VariablesSecureApp vars, String WindowNo,
+      String TableName, String C_DocTypeTarget_ID, String C_DocType_ID, boolean onlyDocType,
+      boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
+      throw new IllegalArgumentException("Utility.getDocumentNo - required parameter missing");
     }
+    final String AD_Client_ID = Utility.getContext(conn, vars, "AD_Client_ID", WindowNo);
+    final String AD_Org_ID = Utility.getContext(conn, vars, "AD_Org_ID", WindowNo);
+
+    final String cDocTypeID = (C_DocTypeTarget_ID.equals("") ? C_DocType_ID : C_DocTypeTarget_ID);
+    if (cDocTypeID.equals("")) {
+      return getDocumentNoByOrg(conn, AD_Client_ID, AD_Org_ID, TableName, updateNext);
+    }
+
+    if (AD_Client_ID.equals("0")) {
+      throw new UnsupportedOperationException("Utility.getDocumentNo - Cannot add System records");
+    }
+
+    CSResponse cs = null;
+    try {
+      cs = DocumentNoData.nextDocType(conn, cDocTypeID, AD_Client_ID, (updateNext ? "Y" : "N"));
+    } catch (final ServletException e) {
+    }
+
+    if (cs == null || cs.razon == null || cs.razon.equals("")) {
+      if (!onlyDocType) {
+        return getDocumentNoByOrg(conn, AD_Client_ID, AD_Org_ID, TableName, updateNext);
+      } else {
+        return "0";
+      }
+    } else {
+      return cs.razon;
+    }
+  }
+
+  @Override
+  public String getDocumentNo(Connection conn, ConnectionProvider con, VariablesSecureApp vars,
+      String WindowNo, String TableName, String C_DocTypeTarget_ID, String C_DocType_ID,
+      boolean onlyDocType, boolean updateNext) {
+    if (TableName == null || TableName.length() == 0) {
```

Comment:
As "TableName" is a String. Is it not better to use 
 `StringUtils.isBlank(TableName) ?`